src/ray/raylet/worker_pool.cc

```diff
@@ -664,15 +664,23 @@ void WorkerPool::PushWorker(const std::shared_ptr<WorkerInterface> &worker) {
   RAY_CHECK(worker->GetAssignedTaskId().IsNil())
       << "Idle workers cannot have an assigned task ID";
   auto &state = GetStateForLanguage(worker->GetLanguage());
-  auto it = state.dedicated_workers_to_tasks.find(worker->GetProcess());
-  if (it != state.dedicated_workers_to_tasks.end()) {
-    // The worker is used for the actor creation task with dynamic options.
-    // Put it into idle dedicated worker pool.
-    const auto task_id = it->second;
+  auto it_p = state.pending_dedicated_workers_to_tasks.find(worker->GetProcess());
+  auto it_r = state.registered_dedicated_workers_to_tasks.find(worker->GetProcess());
+  if (it_p != state.pending_dedicated_workers_to_tasks.end()) {
+    // The worker is used for a task which needs a dedicated worker process.
+    // Put it into the idle dedicated worker pool.
+    const auto task_id = it_p->second;
```

Comment:
why don't we need to remove the entry from `pending_dedicated_workers_to_tasks` here?

Reply:
This is the existing code, and the entries are removed in PopWorker here: https://github.com/ray-project/ray/pull/15545/files/dc9402c85dc504013ce8ce3522541379f74136cd#diff-4c19d6ebd8575189e9439fa5646b567546dc54f64d576a959ce1bb43902d1a03L842

But it's a good question, I'm not really sure why they need to be removed in PopWorker and not in PushWorker.  I'll keep thinking about it

Reply:
Okay, I think the two approaches are actually equivalent, so let me move the erasing from Popworker to Pushworker.  This will also allow me to unify the two maps to address your comment above https://github.com/ray-project/ray/pull/15545/files/dc9402c85dc504013ce8ce3522541379f74136cd#r621478246.